canopycover_readme.md

```diff
@@ -0,0 +1,52 @@
+**Extractor name: Canopy cover extractor**
+
+**Date:10-18-2017**
+
+**Author:Zongyang Li, Maxwell Burnette, Robert Pless**
+
+**Extractor Description: This extractor processes binary stereo images and generates plot-level canopy cover percentage traits for BETYdb.**
+
+**Inputs and Outputs:**
+
+Inputs: RGB raw data, two .bin files that from left and right camera
+
+Reference metadata in .json file
+
+Outputs:  a percentage number represent for how much is the plant in the area.
+
+                           		   
+
+**Algorithm description**
+
+The core idea for this extractor is a plant-soil segmentation. We apply a threshold to differentiate plant and soil, and do a smoothing after binary processing. At last it returns a plant area ratio within the bounding box.
+
+Steps:
+
+1. Split image data into R,G,B channel, and make a tmp image.
+
+2. For each pixel, if G value is T(threshold) higher than R value, make this pixel as foreground, and set the tmp pixel value to 255, so all tmp pixels are 0 or 255.
+
+3. Use a filter to blur this tmp image,
+
+4. Threshold the blurred tmp image with a threshold of 128 to get a new mask image that represents our plant (foreground) detections.
+
+5. Output ratio = foreground pixel count / total pixel count
+
+**Parameters:**
+
+	G - R Threshold is set to 5 for normal situation.
+
+	Blur image to new mask threshold is set to 128
+
+**Failure Conditions:**
+
+1. Image captured in a low illumination.
+
+2. Image captured in a very high illumination.
+
+3. In late season, panicle is covering a lot in the image, and leaves are getting yellow.
+
+4. Sometimes sensor has problem in giving a reasonable image.
```

Comment:
* For each of these points, what is the failure? Is it that the mask fails to accurately separate plant and soil?
* For this point 4, please clarify 'sometimes', 'problem', and 'reasonable'
   * for sometimes: under what conditions ... is this under conditions other than low or high illumination or yellow panicles? or other conditions? Or is it for unidentified reasons

Here are some examples of how I would rewrite these if I knew the answers: 

1. When the image is captured in low illumination, the algorithm fails to detect all leaves as plant material.
2. When the image captured in a very high illumination ...
    (for 1 and 2 define 'low' or 'high' if this can be done in terms of solar flux or image contrast)
3. In late season, when a panicle is covering a lot in the image and (/ or?) leaves are getting yellow, the algorithm fails to correctly classify the panicle or yellow leaf correctly as plant. This issue should be of lower concern as it occurs long after canopy closure and peak growth.
4. ~~Sometimes sensor has problem in giving a reasonable image.~~ Under condition X, the cameras fail to capture expected images of plants or soil. In some cases the result is a blank image.

The examples in https://github.com/terraref/reference-data/issues/186#issuecomment-333631648 are great, although the sample images show what a bad image looks like rather than how the algorithm fails (not required, but could be nice)

Reply:
In addition to the referenced comment, are there any proposed solutions to resolve any of these ... e.g. to handle a wider range of illumination, or to avoid and/or check for the failures?